### PR TITLE
feat(transformItems): Add `transformItems` function to `useRecommendations` hook

### PR DESCRIPTION
**Pull request by shortcuts**
**Summary**

Similar API as in [InstantSearch.js](https://www.algolia.com/doc/api-reference/widgets/hits/js/#widget-param-transformitems), it receive the `items` returned by the query and allow the user to change its content, order, etc. The shape returned should remain the same.

**Results**

Any `recommendations` component can now use `transformItems` to transform the results returned by Algolia. 

**Preview**
<p align="center">
  
</p>

### CODE REVIEW

File: packages/react-recommendations/src/useRecommendations.ts [hunk 1]
```diff
@@ -23,15 +24,16 @@ export type UseRecommendationsProps = {
   maxRecommendations?: number;
   searchParameters?: SearchOptions;
   threshold?: number;
+  transformItems?: RecommendationTransformItems<TObject>;
```
Comment (francoischalifour) [inline packages/react-recommendations/src/useRecommendations.ts]: We can inline `RecommendationTransformItems` if we only use it here.

**Review by francoischalifour — APPROVED** [review]
Let's merge this once https://github.com/algolia/react-recommendations/pull/9/files#r632578219 is addressed, if it makes sense to you.

File: packages/react-recommendations/src/types/RecommendationTransformItems.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export type RecommendationTransformItems<TObject> = (
```
Comment (Haroenv) [inline packages/react-recommendations/src/types/RecommendationTransformItems.ts]: why is this not called `TransformItems`, as it's not related to recommendations

Comment (francoischalifour) [reply packages/react-recommendations/src/types/RecommendationTransformItems.ts]: We can inline it so we don't have to name it.

File: packages/react-recommendations/src/useRecommendations.ts (hunk 1 above)
Comment (shortcuts) [reply packages/react-recommendations/src/useRecommendations.ts]: Sure!

File: packages/react-recommendations/src/types/RecommendationTransformItems.ts (hunk 2 above)
Comment (shortcuts) [reply packages/react-recommendations/src/types/RecommendationTransformItems.ts]: Done in https://github.com/algolia/react-recommendations/pull/9/commits/c4e34b585b80325f2f55bef15c2d68451683c7cb

File: packages/react-recommendations/src/useRecommendations.ts (hunk 1 above)
Comment (shortcuts) [reply packages/react-recommendations/src/useRecommendations.ts]: Done in https://github.com/algolia/react-recommendations/pull/9/commits/c4e34b585b80325f2f55bef15c2d68451683c7cb

**Review by Haroenv — APPROVED** [review]

**Review by francoischalifour — APPROVED** [review]

**Pull request merged**